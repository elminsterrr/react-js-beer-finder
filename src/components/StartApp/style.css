.StartApp-main {
  margin-top: 20px;
}
