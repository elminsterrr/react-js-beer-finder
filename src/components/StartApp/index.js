import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import './style.css';

import BeerListingEntry from '../../containers/BeerListingEntry';

class StartApp extends Component {
  constructor() {
    super();

    this.state = {
      startWasClicked: false,
    };
  }

  handleStartApp = () => {
    this.setState({ startWasClicked: true });
  };

  render() {
    return (
      <div>
        <div className="StartApp-container">
          {!this.state.startWasClicked && (
            <RaisedButton
              label="Start Here"
              className="StartApp-main"
              onClick={this.handleStartApp}
            />
          )}
        </div>
        {this.state.startWasClicked && <BeerListingEntry />}
      </div>
    );
  }
}

export default StartApp;
