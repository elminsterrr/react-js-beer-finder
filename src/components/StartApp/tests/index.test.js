import React from 'react';
import { shallow } from 'enzyme';
import StartApp from '../';

describe('StartApp', () => {
  const startApp = shallow(<StartApp />);

  it('renders correctly', () => {
    expect(startApp).toMatchSnapshot();
  });

  describe('when clicking `Start Here` button', () => {
    startApp.find('.StartApp-main').simulate('click');

    it('changes `state().startWasClicked` to `true`', () => {
      expect(startApp.state().startWasClicked).toBeTruthy();
    });

    it('hides `Start Here` button', () => {
      expect(startApp.find('.StartApp-container').children().length).toEqual(0);
    });
  });
});
