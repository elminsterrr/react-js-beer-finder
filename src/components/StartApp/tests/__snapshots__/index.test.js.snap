// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartApp renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartApp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="StartApp-container"
/>,
        <Connect(BeerListingEntry) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "StartApp-container",
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="StartApp-container"
/>,
          <Connect(BeerListingEntry) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "StartApp-container",
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
